### PR TITLE
CSP compliant docs

### PR DESCRIPTION
**Pull request by aayushman-singh**
This PR addresses #12498 by updating the documentation to align with CSP compliance.

### CODE REVIEW

**Comment by aayushman-singh** [comment]
Working on the three other required changes as well, will mark ready to merge when I'm finished!

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (gasman) [inline docs/advanced_topics/documents/title_generation_on_upload.md:46]: I don't think putting a Django template tag inside a string like this is going to work - it'll just output the tag verbatim, rather than evaluating it. Please test this code.

There's an example of outputting a `<script src="...">` tag from a hook here: https://docs.wagtail.org/en/stable/reference/hooks.html#insert-editor-js

**Review by gasman — CHANGES_REQUESTED** [review]

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (aayushman-singh) [reply docs/advanced_topics/documents/title_generation_on_upload.md:46]: Will do, thanks!

**Comment by aayushman-singh** [comment]
Please check at your convenience @gasman, static keyword has been made explicit and title gen docs for files and images has been merged. Need feedback if changes to methods are needed or if I should continue to keep them seperate.

**Comment by aayushman-singh** [comment]
Tested changes locally as well

File: docs/advanced_topics/images/title_generation_on_upload.md
Comment (RealOrangeOne) [inline docs/advanced_topics/images/title_generation_on_upload.md:1]: Question: Is it intentional to delete this file? I realise merging it with the documents one removes some duplication, but as the file is still under `documents/`, it could be confusing to people specifically looking for image title generation.

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (aayushman-singh) [reply docs/advanced_topics/documents/title_generation_on_upload.md:33]: My mistake, I'll add it back. Thanks for pointing it out.

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (aayushman-singh) [reply docs/advanced_topics/documents/title_generation_on_upload.md:48]: Not familiar, but I'll look into it. Might need help deciding a direction for this.

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (aayushman-singh) [reply docs/advanced_topics/documents/title_generation_on_upload.md:51]: I understand, however I thought it might be easier for people to understand if it was right in front of them as they copied the code. Will remove if needed.

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (aayushman-singh) [reply docs/advanced_topics/documents/title_generation_on_upload.md:57]: let is favoured more nowadays, var works for both older and newer versions so I thought I'd use it since people aren't always upto date. Of course, if I'm allowed to change the code, I'll put let in.

File: docs/advanced_topics/images/title_generation_on_upload.md
Comment (lb-) [reply docs/advanced_topics/images/title_generation_on_upload.md:1]: No. It's not meant to be deleted, sorry if the issue confused you. 

It's meant to be updated along with the other title update page for documents.

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (lb-) [inline docs/advanced_topics/documents/title_generation_on_upload.md:1]: As per other comments, we need to keep these two oages, yes they a similar but what I mean is that the updates to the script usage snippets will also be similar.

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (lb-) [reply docs/advanced_topics/documents/title_generation_on_upload.md:57]: Let's use `const`. It's going to work all the way back to IE11 so we're safe. 

https://caniuse.com/?search=const

I'm not sure why we originally used `var` I wrote this but maybe it was just to be 'careful'. IE11 is now deprecated anyway so `const` is a safe option.

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (lb-) [reply docs/advanced_topics/documents/title_generation_on_upload.md:48]: Good call, let's use that `format_html` for other code snippet changes also. 

The mark_safe was used to make it so that we didn't have to escape the JS code. 

https://docs.djangoproject.com/en/5.1/ref/utils/#django.utils.html.format_html

File: docs/extending/admin_views.md
Comment (lb-) [inline docs/extending/admin_views.md:108]: Just watch the formatting here, it's a bit out of wack. We don't run prettier on our docs but I think this can be a bit cleaner.

File: docs/extending/admin_views.md
Comment (lb-) [inline docs/extending/admin_views.md:113]: We don't need this, for those that know how CSP works they will know this. 

For those that don't care or know about CSP they will just blindly copy the example and use the more secure approach. 

Where possible we want our docs to be simple and not add cognitive overhead or give people too many options/decisions to make.

File: docs/extending/admin_views.md
Comment (lb-) [inline docs/extending/admin_views.md:201]: Thanks. Nice find.

Comment (aayushman-singh) [reply docs/extending/admin_views.md:201]: Thank you

File: docs/extending/admin_views.md
Comment (aayushman-singh) [reply docs/extending/admin_views.md:113]: I understand, thanks

File: docs/extending/admin_views.md
Comment (aayushman-singh) [reply docs/extending/admin_views.md:104]: Done, all left aligned with proper indentation. I'll commit the changes

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (aayushman-singh) [reply docs/advanced_topics/documents/title_generation_on_upload.md:1]: I'll be reverting this change as the issue did not want a merge

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (aayushman-singh) [reply docs/advanced_topics/documents/title_generation_on_upload.md:48]: Alright, I'll use `format_html` for all the methods.

File: docs/advanced_topics/documents/title_generation_on_upload.md
Comment (aayushman-singh) [reply docs/advanced_topics/documents/title_generation_on_upload.md]: I've removed them all with just the general note at the top

File: docs/advanced_topics/images/title_generation_on_upload.md
Comment (lb-) [reply docs/advanced_topics/images/title_generation_on_upload.md:1]: There is no need to reduce redundancy here, we just need the code examples updated to a CSP compliant method.

**Comment by aayushman-singh** [comment]
Thank you for your patience, I'll complete all the changes and ask for a review soon. Could I ask for some guidance on what you meant by the third point of the issue @lb- ?

File: docs/advanced_topics/images/title_generation_on_upload.md
Comment (aayushman-singh) [reply docs/advanced_topics/images/title_generation_on_upload.md:1]: Understood, thanks!

**Comment by aayushman-singh** [comment]
I've completed all the points on the issue now incorporating all the feedback, please let me know if anything else is needed. 
I went and replaced `mark_safe` usage in the `docs/reference/hooks.md` with `format_html` since I caught them but this was not mentioned in the issue originally so let me know if it needs to be reverted.

File: docs/contributing/documentation_guidelines.md
Comment (lb-) [inline docs/contributing/documentation_guidelines.md:443]: Great start here, I will clean up the formatting a bit but I think this is a solid start.

Other core team members can work on this over time.

File: docs/contributing/documentation_guidelines.md
Comment (lb-) [inline docs/contributing/documentation_guidelines.md:454]: I'll leave these out for now but great suggestions, maybe something to consider for the future.

We may even want to say something like 'remember the AI is watching :D and example code will become the code of the future' haha.

File: docs/extending/admin_views.md
Comment (lb-) [inline docs/extending/admin_views.md:99]: I think this needs to be wrapped in code blocks but I can do when merging if needed.

File: docs/reference/hooks.md
Comment (lb-) [inline docs/reference/hooks.md:515]:
```suggestion
```html+django
```

File: docs/extending/admin_views.md
Comment (lb-) [inline docs/extending/admin_views.md:100]:
```suggestion
```css
/* calendar.css */
```

File: docs/contributing/documentation_guidelines.md
Comment (lb-) [inline docs/contributing/documentation_guidelines.md:454]: I'll leave these out for now but great suggestions, maybe something to consider for the future.

We may even want to say something like 'remember the AI is watching :D and example code will become the code of the future' haha.

```suggestion
```

**Review by lb- — APPROVED** [review]
Last few changes are mostly formatting and a few nitpicks, other than that it looks great.

Thank you @aayushman-singh  I will clean up the branch and push shortly.

File: docs/reference/hooks.md
Comment (lb-) [inline docs/reference/hooks.md:521]: @aayushman-singh I think we probably should keep this as the non-Stimulus example. I know I suggested this but I realise there's a bit more to it compared to what's in this example.

I will just move the existing JS to the external file and push.

File: docs/reference/hooks.md
Comment (lb-) [inline docs/reference/hooks.md:130]: Best to avoid changes to other areas of the docs, this may not be the nicest formatting but if you think it's something to fix, please can you prepare a separate PR with just this change.

Comment (lb-) [reply docs/reference/hooks.md:130]: Actually, sorry I realise you changed it to format_html, ignore my comment. let's add this.

**Comment by aayushman-singh** [comment]
Thanks for all your help and patience, I'll get better and make more meaningful contributions in the future!

**Pull request merged**
